core/src/replay_stage.rs

```diff
@@ -3518,8 +3525,8 @@ impl ReplayStage {
                 total_threshold_stake,
                 total_epoch_stake,
             ) = {
-                let fork_stats = progress.get_fork_stats(bank.slot()).unwrap();
-                let propagated_stats = &progress.get_propagated_stats(bank.slot()).unwrap();
+                let fork_stats = progress.get_fork_stats(check_slot).unwrap();
+                let propagated_stats = &progress.get_propagated_stats(check_slot).unwrap();
```

Comment:
I'm confused: is_locked_out and others will be used later in deciding where to vote? So why change the slot here?

Reply:
if we are voting `check_slot` is equal to `bank.slot()`. See the `assert_eq` on line 3569.

Reply:
I meant, what will happen if you use bank.slot() here but replace with check_slot in all the failure_reasons from line 3547~3558?

Also, should we report both banks in the failure reason?

Reply:
> I meant, what will happen if you use bank.slot() here but replace with check_slot in all the failure_reasons from line 3547~3558?

I think it would be even more confusing, because then you'll still perform the checks on the reset bank, but now report them under the heaviest bank.

> Also, should we report both banks in the failure reason?

i don't see a compelling reason to perform and log checks on a reset bank